sklearn/svm/src/liblinear/liblinear_helper.c

```diff
@@ -204,8 +203,7 @@ double get_bias(struct model *model)
 
 void free_problem(struct problem *problem)
 {
-    int i;
-    for(i=problem->l-1; i>=0; --i) free(problem->x[i]);
```

Comment:
please summarise why this is not needed.

Reply:
`problem->x[1]` through `problem->x[l-1]` are just pointers into the middle of the array that was allocated at `problem->x[0]`. The call to `free(problem->x[0])` frees the entire array. Do you think I should add an explanatory comment here?